views.py

```diff
@@ -68,22 +69,26 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
 
     resource_cls = check_resource_type(res.resource_type)
 
+    if federated_path:
+        res_root = os.path.join(federated_path, res_id)
+    else:
+        res_root = res_id
+
+    bag_modified = "false"
+    metadata_dirty = "false"
+    if istorage.exists(res_root):
+        bag_modified = istorage.getAVU(res_root, 'bag_modified')
+        metadata_dirty = istorage.getAVU(res_root, 'metadata_dirty')
+
```

Comment:
Since ```bag_modified``` get reset only when a bag is recreated (upon user requesting bag download), we can't use that flag to determine if the xml file regeneration is needed or not. Say after making some changes to metadata, the user wants to download the science metadata xml file (via REST API). If we only depend on ```bag_modified``` then that file will be regenerated every time the user tries to get that file until a bag download happens for that resource.